src/frontend/src/binder/statement.rs

```diff
@@ -28,6 +29,7 @@ pub enum BoundStatement {
     Delete(Box<BoundDelete>),
     Update(Box<BoundUpdate>),
     Query(Box<BoundQuery>),
+    CreateView(Box<BoundCreateView>),
```

Comment:
After reading the whole PR, it seems unavoidable to add `BoundCreateView` to support the prepared statement, however, it also means if we want to support more statement types for the prepared statement more BoundXXX needed to be added.  @xiangjinwu what do you think

Reply:
I tried to reuse `BoundQuery` instead of introducing `BoundCreateView` in https://github.com/risingwavelabs/risingwave/pull/17769/commits/e01e9140cbcfd18fe31430002f5cfc0bd3157cc9 .

It seems to work, but

1. It was using the AST `Statement::CreateView` instead of a bound query i.e. `BoundCreateView` ([here](https://github.com/risingwavelabs/risingwave/commit/e01e9140cbcfd18fe31430002f5cfc0bd3157cc9#diff-8116a957c88da910eda3c8acb95b432248284ec7999ccc6b9d59ac104382be4cR110-R118)), which can work but different from existing design.
2. I feel it makes code more confusing that both `create materialized view` and `select` are converted to `BoundQuery` here.